runner/utils/driver/istio/helpers.go

```diff
@@ -257,6 +260,76 @@ func prepareVirtualServiceHTTPRoutes(ctx context.Context, projectID, serviceID s
 	return httpRoutes, nil
 }
 
+func prepareVirtualServiceTCPRoutes(ctx context.Context, projectID, serviceID string, services map[string]model.AutoScaleConfig, routes model.Routes, proxyPort uint32) ([]*networkingv1alpha3.TCPRoute, error) {
+	var tcpRoutes []*networkingv1alpha3.TCPRoute
+
+	for _, route := range routes {
+		if route.Protocol != model.TCP {
+			continue
+		}
+		// Check if the port provided is correct
+		if route.Source.Port == 0 {
+			return nil, errors.New("port cannot be zero")
+		}
+
+		// Check if at least one target is provided
+		if len(route.Targets) == 0 {
+			return nil, errors.New("at least one target needs to be provided")
+		}
+
+		// Prepare an array of targets / destinations
+		var destinations []*networkingv1alpha3.RouteDestination
+		for _, target := range route.Targets {
+			switch target.Type {
+			case model.RouteTargetVersion:
+				// Check if config for version exists
+				versionScaleConfig, p := services[target.Version]
+				if !p {
+					return nil, helpers.Logger.LogError(helpers.GetRequestID(ctx), fmt.Sprintf("version (%s) not found for service (%s)", target.Version, serviceID), nil, nil)
+				}
+
+				// Prepare variables
+				destHost := getInternalServiceDomain(projectID, serviceID, target.Version)
+				destPort := uint32(target.Port)
+
+				// Redirect traffic to runner when no of replicas is equal to zero. The runner proxy will scale up the service to service incoming requests.
+				if versionScaleConfig.MinReplicas == 0 {
+					destHost = "runner.space-cloud.svc.cluster.local"
+					destPort = proxyPort
+				}
```

Comment:
Runner can only handle http requests. Always direct to service. You dont really need versionScaleConfig for tcp.